### PR TITLE
[Release workflow] Workaround for the ownership bug

### PR DESCRIPTION
**Pull request by LukeMathWalker**
## Motivation and Context
The [release job for 0.54.2 failed](https://github.com/awslabs/smithy-rs/actions/runs/4106453596/jobs/7085205295) because we encountered an error when trying to remove owners from our crates.  
This is unfortunately related to an upstream crates.io bug. This PR introduces a patch to work around the issue.

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: tools/ci-build/publisher/src/subcommand/publish.rs [hunk 1]
```diff
@@ -162,21 +172,26 @@ pub async fn correct_owner(handle: &PackageHandle, category: &PackageCategory) -
...
+                }
                 // Adding an individual owner requires accepting an invite, so don't attempt to remove
                 // anyone if an owner was added, as removing the last individual owner may break.
                 // The next publish run will remove the incorrect owner.
                 if !added_individual {
-                    cargo::RemoveOwner::new(&handle.name, incorrect_owner)
+                    cargo::RemoveOwner::new(&handle.name, crate_owner)
                         .spawn()
                         .await
-                        .context(format!("remove incorrect owner `{}` from crate `{}`", incorrect_owner, handle))?;
```
Comment (jjant) [inline tools/ci-build/publisher/src/subcommand/publish.rs:173]: I assume this call failed in [this run](https://github.com/awslabs/smithy-rs/actions/runs/4106453596/jobs/7085205295#step:8:364). Why didn't we get this error message in the logs?

Comment (jjant) [reply tools/ci-build/publisher/src/subcommand/publish.rs:173]: Also you may want `.with_context` instead to avoid allocating the string.

**Review by jjant — APPROVED** [review]
While you're here, you may want to [fix this warning](https://github.com/awslabs/smithy-rs/actions/runs/4106453596/jobs/7085205295#step:8:324):

```diff
- hashes.push_str(&sha256::digest_file(import).context("hash model")?);
+ hashes.push_str(&sha256::try_diggest(import).context("hash model")?);
```

File: tools/ci-build/publisher/src/subcommand/publish.rs (hunk 1 above)
Comment (LukeMathWalker) [reply tools/ci-build/publisher/src/subcommand/publish.rs:173]: Done in https://github.com/awslabs/smithy-rs/pull/2318/commits/28ee6f3c7ae7fc53818439c2badcb6042469ec8d
I'm looking into the logs 🤔

Comment (LukeMathWalker) [reply tools/ci-build/publisher/src/subcommand/publish.rs:173]: We got the log for each attempt (`2023-02-06T18:24:58.190589Z  INFO publisher::retry: Correcting ownership of `aws-smithy-async` failed on attempt 1 with retryable error: remove incorrect owner `github:awslabs:smithy-rs-server` from crate `aws-smithy-async-0.54.2`. Will retry after 5s`), but it lacks the inner details. I have fixed it in https://github.com/awslabs/smithy-rs/pull/2318/commits/fa9afc72632cd33eedb1e44750e6fd391a3f7462

Comment (jjant) [reply tools/ci-build/publisher/src/subcommand/publish.rs:173]: Great ⭐

**Review by jdisanti — APPROVED** [review]

**Review by crisidev — APPROVED** [review]

**Pull request merged**